### PR TITLE
Shape list

### PR DESCRIPTION
**Pull request by CapsRePeaT**
1. fixed compilation
2. added option to choose between 2d and 3d file loading
3. added functionality of remembering dirs in file dilog when choose second file

### CODE REVIEW

File: src/common/include/common.h [hunk 1]
```diff
@@ -5,9 +5,31 @@
...
-enum Axis { X, Y, Z };
+enum class DrawMode { 
+	Gradient = 0, 
+	Stratified 
+};
+
+enum class ShapeType { 
+	Box = 0, 
+	Sphere, 
+	count,
```
Comment (CapsRePeaT) [inline src/common/include/common.h:16]: Count (upper)

File: src/common/include/common.h [hunk 2]
```diff
@@ -5,9 +5,31 @@
...
+	D3 = 0, 
+	D2, 
+	Undefined
+};
+
+enum Axis { 
+	X = 0, 
+	Y, 
+	Z, 
+	count,
```
Comment (CapsRePeaT) [inline src/common/include/common.h:30]: Count (upper)

File: src/core/src/core.cpp [hunk 3]
```diff
@@ -3,20 +3,41 @@
...
 	// TODO: If we create new file representation here,
 	// how can reader set its id to shapes?
 	// TODO: BTW, we assign objects to getters here.
 	//  1. We need setter for this operation
 	//  2. Copy of 2 huge objects. Should be using pointers/references 
 	//     or anything else to avoid it
-	new_representation.geom_storage() = reader.geometry();
-	new_representation.heatmaps()     = reader.heatmaps();
+	//new_representation.geom_storage() = reader.geometry();
+	//new_representation.heatmaps()     = reader.heatmaps();
```
Comment (CapsRePeaT) [inline src/core/src/core.cpp:41]: delete

File: src/core/include/core.h [hunk 4]
```diff
@@ -16,7 +16,9 @@ class Core {
 		return instance;
 	}
 	void LoadHeatmap(std::string file_path_mv) {}
-	void LoadGeometry(std::string trm_file_path_mv, std::string t2d_file_path_mv);
+	void LoadGeometry(std::string trm_file_path_mv, 
+									  std::string t2d_file_path_mv,
```
Comment (Saiel) [inline src/core/include/core.h:20]: Do not use tabs for aligning. Thery are only for indent, that is level 1 here. The rest should be spaces

File: src/core/src/core.cpp [hunk 5]
```diff
@@ -3,20 +3,41 @@
 #include <iostream>
 
 #include "../../readers/src/solver3d/solver3d_reader.h"
+#include "../../readers/src/solver2d/solver2d_reader.h"
 
 void Core::LoadGeometry(std::string trm_file_path_mv,
-                        std::string t2d_file_path_mv) {
+                        std::string t2d_file_path_mv, 
+												const GeometryType type) {
```
Comment (Saiel) [inline src/core/src/core.cpp:10]: Same as in header. Also trailing space on 9th line

File: src/ui/src/mainwindow.h [hunk 6]
```diff
@@ -20,18 +20,21 @@ class MainWindow final : public QMainWindow {
 	explicit MainWindow(QWidget* parent = nullptr);
 	~MainWindow() override;
 	void LoadFile(std::string trm_file_path, 
-								std::string t2d_file_path);
+								std::string t2d_file_path,
```
Comment (Saiel) [inline src/ui/src/mainwindow.h:23]: Same as in `core.h`.
Trailing space

**Review by Saiel — CHANGES_REQUESTED** [review]
Tab aligning must be fixed before merging, sorry. The rest seems ok

File: src/ui/src/mainwindow.cpp [hunk 7]
```diff
@@ -62,17 +65,28 @@ void MainWindow::LoadFile(std::string trm_file_path,
...
-	const QString t2d_file = QFileDialog::getOpenFileName(
-			this, tr("Open T2D File"), QDir::currentPath(),
-			tr("geom (*.txt *.T2D);; ALL (*.*)"));
-	// TODO: check if file_name is empty (on cancel)
+	if (trm_file.isEmpty())
+		return;
+	QDir trm_dir = QFileInfo(trm_file).absoluteDir();
+	QString absolute_trm_dir = trm_dir.absolutePath();
+	const QString t2d_file = QFileDialog::getOpenFileName(this, tr("Open T2D File"),
+																												absolute_trm_dir,
```
[the remark this answers is not in the document]
Comment (Saiel) [reply src/ui/src/mainwindow.cpp:85]: 84th line exceeds line size limit

**Review by Saiel — CHANGES_REQUESTED** [review]

File: src/ui/src/mainwindow.cpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (CapsRePeaT) [reply src/ui/src/mainwindow.cpp:85]: 80 not a strict limit, if you go a bit more but improve readibility it's ok

**Review by Saiel — APPROVED** [review]

**Pull request merged**